### ISSUE TITLE
Nomad Scheduler Support

### ISSUE THREAD

**Issue by mwlon** [meta-pytorch/torchx#734]
## Description
Add support for [Hashicorp Nomad](https://www.nomadproject.io/) as a scheduler.

## Motivation/Background
Nomad has a good scheduler, and pytorch has good distributed training. However, Nomad launches batch job tasks asynchronously instead of all at once, so torchx will be necessary to train in an orderly way when not all resources are immediately available.

## Proposal
Add a [torchx.schedulers.Scheduler](https://pytorch.org/torchx/latest/schedulers.html#scheduler-classes) subclass.

**Comment by kiukchung** [comment]
Sounds reasonable. Would you be open to contributing the Nomad scheduler integration? Happy to review the PR. Thanks!

**Comment by kiukchung** [comment]
Closing as not planned: torchx no longer takes new in-tree schedulers. The in-tree set is `local_cwd`, `local_docker`, `kubernetes` and `slurm`. Out-of-tree schedulers now plug in through `torchx.plugins`: decorate a factory with `@register.scheduler()` in a `torchx_plugins.*` namespace package and torchx discovers it (see `docs/source/plugins.rst`).
